internal/diagnostics/logs.go

```diff
@@ -0,0 +1,184 @@
+)
+
+type ModuleLogs struct {
+	Logs map[string][]string
+}
+
+type LogOptions struct {
+	Since time.Duration
+	Lines int64
+}
+
+type ModuleLogsCollector struct {
+	client kube.Client
+	*out.Printer
+}
+
+func NewModuleLogsCollector(client kube.Client) *ModuleLogsCollector {
+	return &ModuleLogsCollector{
+		client:  client,
+		Printer: out.Default,
+	}
+}
+
+// RunSince collects error logs from specified modules for a given time duration
+func (c *ModuleLogsCollector) RunSince(ctx context.Context, modules []string, since time.Duration) ModuleLogs {
+	logOptions := LogOptions{Since: since}
+	return c.collectLogs(ctx, modules, logOptions)
+}
+
+// RunLast collects the last N lines of error logs from specified modules
+func (c *ModuleLogsCollector) RunLast(ctx context.Context, modules []string, last int64) ModuleLogs {
+	logOptions := LogOptions{Lines: last}
+	return c.collectLogs(ctx, modules, logOptions)
+}
+
+// collectLogs is the generic method that handles both since and last scenarios
+func (c *ModuleLogsCollector) collectLogs(ctx context.Context, modules []string, options LogOptions) ModuleLogs {
+	allLogs := make(map[string][]string)
+
+	for _, module := range modules {
+		c.collectModuleLogs(ctx, module, options, allLogs)
+	}
+
+	return ModuleLogs{Logs: allLogs}
+}
+
+// collectModuleLogs collects error logs from a single module
+func (c *ModuleLogsCollector) collectModuleLogs(ctx context.Context, moduleName string, options LogOptions, allLogs map[string][]string) {
+	pods, err := c.getPodsForModule(ctx, moduleName)
+	if err != nil {
+		out.Verbosefln("Failed to get pods for module %s: %v", moduleName, err)
+	}
+
+	for _, pod := range pods {
+		c.collectPodLogs(ctx, &pod, options, allLogs)
+	}
+}
+
+// getPodsForModule gets all pods for a specific module in kyma-system namespace
+func (c *ModuleLogsCollector) getPodsForModule(ctx context.Context, moduleName string) ([]corev1.Pod, error) {
+	labelSelector := map[string]string{
+		"kyma-project.io/module": moduleName,
+	}
+
+	deploymentsList, err := c.client.Static().AppsV1().Deployments("kyma-system").List(ctx, metav1.ListOptions{
+		LabelSelector: resources.LabelSelectorFor(labelSelector),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list deployments for module %s: %w", moduleName, err)
+	}
+
+	podsList := make([]corev1.Pod, 0)
+
+	for _, deployment := range deploymentsList.Items {
+		matchLabels := deployment.Spec.Selector.MatchLabels
+		pods, err := c.client.Static().CoreV1().Pods("kyma-system").List(ctx, metav1.ListOptions{
+			LabelSelector: resources.LabelSelectorFor(matchLabels),
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to list pods for deployment %s: %w", deployment.Name, err)
+		}
+
+		podsList = append(podsList, pods.Items...)
+	}
+
+	return podsList, nil
+}
+
+// collectPodLogs collects error logs from a pod
+func (c *ModuleLogsCollector) collectPodLogs(ctx context.Context, pod *corev1.Pod, options LogOptions, allLogs map[string][]string) {
```

Comment:
The same as before.
I also see an inconsistency with `c.collectContainerLogs()`.